### DISCUSSION TITLE
turbo build for @preconstruct/monorepo

### DISCUSSION THREAD

**Discussion by rake7h** [vercel/turborepo#564] [Help]
I'm trying to figure out a way to run the [preconstruct](https://github.com/preconstruct/preconstruct) command through turbo pipeline on lerna & yarn workspace. 

my script:

```js
{
"build":  "turbo run preconstruct build",
}
```

pipeline

```js
"build": {
"dependsOn": ["^build"]
},
```

I tried above config, but it didn't work.

Some example repos with preconstruct build script

[https://github.com/tmm/wagmi/blob/main/package.json#L4](https://github.com/tmm/wagmi/blob/main/package.json#L4 )
[https://github.com/chakra-ui/chakra-ui/blob/main/package.json#L43](https://github.com/chakra-ui/chakra-ui/blob/main/package.json#L43
)

any idea how can i enable `yarn preconstruct build` with turbo pipeline?

**Comment by anthonyshew** [comment]
Old question but for anyone searchers:

The expression `turbo run preconstruct build` says "Run my preconstruct and build scripts with as much parallelism as possible."

It's likely that the intent here was to create "preconstruct" before "build" which would be:

```
# package.json
{
  "preconstructor": "preconstruct build",
  "build": "<build task>"
}
```

```
# turbo.json
"build": {
  "dependsOn": ["^build", "preconstruct"]
},

```

**Reply by rake7h** [reply]
Well, `preconstruct build` is a single command which triggers the build of [preconstruct](https://preconstruct.tools/commands#build) for all workspace packages belongs to a monorepo. I believe it's not possible to run this command per package and leverage the incremental caching and parallelism.


preconstruct is not design to support the build command per packages. It uses Rullup internally to build the whole monorepo in single go in a loop via `preconstruct build` at root package.json script.
